### PR TITLE
chore: Text Editor hooks.

### PR DESCRIPTION
**Pull request by richburdon**

### CODE REVIEW

File: packages/ui/react-ui-editor/src/components/TextEditor/TextEditor.stories.tsx [hunk 1]
```diff
@@ -57,19 +56,5 @@ export default {
...
-
-export const Text = {
   render: () => <Story text={initialText} placeholder='Enter text...' />,
 };
-
-export const Automerge = {
-  render: () => <Story text={initialText} placeholder='Enter text...' automerge />,
-};
-
-export const Listener = {
```
Comment (wittjosiah) [inline packages/ui/react-ui-editor/src/components/TextEditor/TextEditor.stories.tsx:71]: we should recreate this story somewhere, it's helpful for validating the listener is working as expected. I needed it when getting the local files plugin working again

File: packages/ui/react-ui-editor/src/extensions/markdown/decorate.ts [hunk 2]
```diff
@@ -208,31 +211,35 @@ export const decorateMarkdown = (options: DecorateOptions = {}) => {
...
         }
 
-        clearUpdate() {
-          if (this.pendingUpdate > -1) {
-            window.clearTimeout(this.pendingUpdate);
-            this.pendingUpdate = -1;
-          }
-        }
-
+        // TODO(burdon): BUG: If the cursor is at the end of a link at the end of a line,
```
Comment (wittjosiah) [inline packages/ui/react-ui-editor/src/extensions/markdown/decorate.ts:229]: https://github.com/dxos/dxos/assets/4529818/39b5e2b7-ebd5-440f-b3cd-9f2314959743

I captured a video of what I think you're describing here just yesterday

File: packages/ui/react-ui-editor/src/components/TextEditor/TextEditor.tsx [hunk 3]
```diff
@@ -18,13 +18,14 @@ import React, {
...
+import { DocAccessor } from '@dxos/echo-schema';
 import { log } from '@dxos/log';
 import { useThemeContext } from '@dxos/react-ui';
 import { focusRing } from '@dxos/react-ui-theme';
 import { isNotFalsy } from '@dxos/util';
 
-import { createBasicBundle, createMarkdownExtensions, editorMode } from '../../extensions';
-import { type EditorModel } from '../../hooks';
+import { createMarkdownExtensions, editorMode } from '../../extensions';
+import { createBasicExtensions, createThemeExtensions, type EditorModel } from '../../hooks';
```
Comment (wittjosiah) [inline packages/ui/react-ui-editor/src/components/TextEditor/TextEditor.tsx:28]: seems weird that this is imported from hooks

File: packages/ui/react-ui-editor/src/extensions/markdown/decorate.ts (hunk 2 above)
Comment (richburdon) [reply packages/ui/react-ui-editor/src/extensions/markdown/decorate.ts:229]: i think marijn just fixed this (i merged it yesterday) -- can you check if you still see this in main.

File: packages/apps/plugins/plugin-markdown/src/extensions.tsx [hunk 4]
```diff
@@ -45,6 +45,7 @@ export const getExtensions = ({ settings, document, dispatch }: ExtensionsOption
     // Common.
     //
     decorateMarkdown({
+      selectionChangeDelay: 400,
```
[the remark this answers is not in the document]
Comment (richburdon) [reply packages/apps/plugins/plugin-markdown/src/extensions.tsx:48]: we added a feature to delay expanding link decorations (removing the widget) when you enter them. this lets you quickly nav through a doc with lots of links/images without it immediately triggering all of them, which made it really jerky.

File: packages/ui/react-ui-editor/src/components/TextEditor/TextEditor.tsx (hunk 3 above)
Comment (richburdon) [reply packages/ui/react-ui-editor/src/components/TextEditor/TextEditor.tsx:28]: right. i'll move them once everything settles and i remove the components.

**Pull request merged**